### PR TITLE
Added constraint & template to allow only private cluster.

### LINKED ISSUE #99: Constraint to allow only private gke cluster

### PR DESCRIPTION
**Pull request by joecheuk**
Resolves #99

### CODE REVIEW

File: policies/templates/gcp_gke_private_cluster_v1.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  crd:
+    spec:
+      names:
+        kind: GCPGKEPrivateClusterConstraintV1
+        plural: gcpgkeprivateclusterconstraintsv1        
+      validation:
+        openAPIV3Schema:
+          properties: {}
+  targets:
+    validation.gcp.forsetisecurity.org:
```
Comment (t12g) [inline policies/templates/gcp_gke_private_cluster_v1.yaml]: let's update the target format

File: samples/allow_only_private_cluster.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+apiVersion: constraints.gatekeeper.sh/v1alpha1
+kind: GCPGKEPrivateClusterConstraintV1
+metadata:
+  name: allow_only_private_cluster
+  annotations:
+    benchmark: GKE_HARDENING_GUIDELINE
+spec:
+  severity: high
+  match:
+    gcp:
```
Comment (t12g) [inline samples/allow_only_private_cluster.yaml]: same here... or delete the match block altogether

File: validator/gke_private_cluster.rego [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	check_cluster_private(cluster)
+
+	message := sprintf("Cluster %v is not private.", [asset.name])
+	metadata := {"resource": asset.name}
+}
+
+###########################
+# Rule Utilities
+###########################
+check_cluster_private(cluster) {
```
Comment (t12g) [inline validator/gke_private_cluster.rego]: nit: is it helpful to extract this separate rule? can we just inline it?

File: validator/test/fixtures/gke_private_cluster/constraints/allow_only_private_cluster/data.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+#
+apiVersion: constraints.gatekeeper.sh/v1alpha1
+kind: GCPGKEPrivateClusterConstraintV1
+metadata:
+  name: allow_only_private_cluster
+  annotations:
+    benchmark: GKE_HARDENING_GUIDELINE
+spec:
+  severity: high
+  match:
```
Comment (t12g) [inline validator/test/fixtures/gke_private_cluster/constraints/allow_only_private_cluster/data.yaml]: let's delete the match block here

File: samples/allow_only_private_cluster.yaml (hunk 2 above)
Comment (joecheuk) [reply samples/allow_only_private_cluster.yaml]: updated

File: validator/gke_private_cluster.rego (hunk 3 above)
Comment (joecheuk) [reply validator/gke_private_cluster.rego]: updated

File: validator/test/fixtures/gke_private_cluster/constraints/allow_only_private_cluster/data.yaml (hunk 4 above)
Comment (joecheuk) [reply validator/test/fixtures/gke_private_cluster/constraints/allow_only_private_cluster/data.yaml]: updated

File: policies/templates/gcp_gke_private_cluster_v1.yaml (hunk 1 above)
Comment (joecheuk) [reply policies/templates/gcp_gke_private_cluster_v1.yaml]: What should i update this to?

Comment (t12g) [reply policies/templates/gcp_gke_private_cluster_v1.yaml]: discussed over chat. we'll bulk update the targets in a separate PR

**Pull request merged**